Feature/redux

mattdeann opened the pull request:
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [x] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)


## *Description: What does this PR do?*
Hooks up API to Redux store. Renders hot takes by category. Gets categories whenever a path requiring them is hit.


## *What are the relevant tickets?*
Closes #14 #15 #16 


## *Bugs:*
Something weird with metro bundler and localhost, sometimes need to reload Metro


## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

LukeMason33 committed:
Install Detox and Jest. Get first mock test to pass.

LukeMason33 committed:
Install Detox and Jest. Get first mock test to pass.

mattdeann committed:
Fix merge conflicts

mattdeann committed:
Set up redux for categories

mattdeann committed:
Set up redux for categories

mattdeann committed:
Set up redux for categories

mattdeann committed:
Set up redux for categories

mattdeann committed:
Finish setting up categories

mattdeann committed:
Finish setting up categories

mattdeann committed:
Resolve merge conflicts

mattdeann committed:
Get categories from api and store in redux

mattdeann committed:
Encounter bug with mapDispatchToProps

mattdeann committed:
Display category data from API

mattdeann committed:
Catch up branch

mattdeann committed:
Merge branch 'main' into feature/redux

LukeMason33 committed:
Pull down changes on my machine. Run pod install

mattdeann committed:
Merge swiper functionality

LukeMason33 committed:
Merge branch 'feature/redux' of github.com:yays-or-nays/yays-or-nays-FE into feature/redux

LukeMason33 committed:
Setup hotTakes reducer

LukeMason33 committed:
Setup action to get hotTake. Refactor HotTakes component to function.

LukeMason33 committed:
Get hotTake state to reflect network request data.

mattdeann committed:
Trigger mutations on swipe. Currently 404ed

mattdeann committed:
Update testData ids to mimic backend

mattdeann committed:
Trigger rerender on card stack

mattdeann committed:
Add functionality to nav buttons

LukeMason33 commented on src/Categories/Categories.js:
```diff
@@ -7,51 +6,43 @@ import {
 } from 'react-native';
+import React from 'react';
 import {Redirect, NativeRouter, Route, Link} from 'react-router-native';
+import {useSelector, useDispatch} from 'react-redux';
```
These hooks are the most useful pieces of code that I've personally found so far! Makes connecting the components to our store a breeze. This was some solid troubleshooting!

LukeMason33 commented on src/HotTakes/HotTakes.js:
```diff
@@ -1,146 +1,161 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
```
We can remove these lines since there's no need for these hooks anymore.

LukeMason33 commented on src/HotTakes/HotTakes.js:
```diff
@@ -1,146 +1,161 @@
-  constructor() {
-    super();
-  }
+  console.log(hotTake);
```
This line can be removed as well since we know we are getting the information we need now!

LukeMason33 commented on src/HotTakes/HotTakes.js:
```diff
@@ -1,146 +1,161 @@
+  return (
+    <View style={{flex: 1}}>
+      <CardStack
+        key={hotTake ? hotTake.id : NaN}
```
This line is to ensure that our CardStack will look for a Card with the same key whenever this component is rendered...right?

LukeMason33 committed:
Remove lines 1 and 26

I removed lines:
1 - Had useState and useEffect hooks that we were not using in this file
26 - There was a console.log() that was left behind.

LukeMason33 commented on src/Redux/store.js:
```diff
@@ -0,0 +1,20 @@
+// Redux Imports
+import {createStore, applyMiddleware, combineReducers} from 'redux';
```
Glad we found this applyMiddleware(thunk) method to handle our asynchronous action creators, good problem solving here!

LukeMason33 committed:
Delete unnecessary comments.

I removed some comments  that we did not need anymore.

LukeMason33 commented on src/util.js:
```diff
@@ -0,0 +1,63 @@
+        //   categories: response.data.data.tags,
+        // });
+      } else {
+        console.log('fail');
```
Here we will have to implement some sort of error handling to ensure our users know if something got messed up.

LukeMason33 commented on src/util.js:
```diff
@@ -0,0 +1,63 @@
+        //   categories: response.data.data.tags,
+        // });
+      } else {
+        return response.status;
```
Same for this line here. Definitely need to add some error handling for users.

LukeMason33 reviewed (approved):
All of this looks good Matt! I'm glad we got this troubleshooting out of the way so we can now focus on polishing this app up! Things for future:
- We need to create some error handling for our netowrk requests to display to our users if they do not go through.
Great job on this everybody!
